docs/developer/release-cycle.md

```diff
@@ -0,0 +1,69 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`).
+
+**Note**: For the `v1.3` and `v1.4` release cycles we assume a ~ 4 months (~ 17 weeks) release cycle. This should be revisited after the `v1.4.0` release.
+
+The release cycle can be split up into the following phases:
+
+* Week 1-12: Active development
+    * Alpha releases based on the main branch can be released when necessary
+* Week 13-14:
+    * Beta releases based on the main branch
+    * The beta releases are meant as a preview what the upcoming release will look like
+    * Provider can start adopting the new release based on the beta releases
+    * Breaking changes to providers have to be discussed and approved in the office hours
```

Comment:
Very good point, thank you!

What would be the right process we want to follow?

A first suggestion:
* Create/have an issue describing the change
* Issue has to be surfaced in the office hours
  * So folks in the office hours that are not necessarily following all our issues have a chance to chime in
  * Important to note: Surfacing the issue in office hours does not have to be done by the issue author in person (time zones, ...). The can also just add it to the agenda and then one of the maintainers can bring it up.
* Issue has to be approved via +1 on the issue

Not sure what the threshold for approval should be given we are talking about breaking changes for providers and it's hard to define what approval "by providers" means (i.e. which provider are we asking, ...). Maybe to keep it simple >= 2 +1s from core CAPI maintainers + no veto from other maintainers?

Open to all suggestions, especially for simpler processes ;)

Reply:
The above makes sense to me, maybe if we want to keep it simple for starters we can say something like `Breaking changes to providers have to be discussed in the office hours and in the absence of clear immediate consensus they'd need to follow an async approval process via issue discussion.`

Reply:
+1 to keep it simple

Reply:
> don't think is the appropriate venue though for making decisions as approve/reject something which I think it should follow an async procedure most of the time

great point, thanks for bringing that up @enxebre 

IMO we should not ever make any consensus during office hours because it's a non-inclusive format. We can discuss in office hours to bring clarity and offer an opportunity for people to ask questions, but consensus should be reached asynchronously. Especially in this case since it touches providers and we don't want a breaking change that affects a provider to be made because there was "clear immediate consensus" in office hours even though the affected provider(s) weren't represented. 

I propose we amend to:

```
Breaking changes to providers should be announced in the provider updates section of the office hours meeting notes and approved in the PR or issue by both approvers and key affected providers.
```

Reply:
I merged this suggestion with Fabrizio's 
WDYT?

>     * All changes that impact providers' adoption of the new release should be announced in the provider updates section 
      of the office hours meeting notes and approved in the PR or issue by both approvers and key affected providers.

Reply:
LGTM